dot/sync/syncer.go

```diff
@@ -306,7 +294,7 @@ func (s *Service) handleBody(body *types.Body) error {
 
 // handleHeader handles blocks (header+body) included in BlockResponses
 func (s *Service) handleBlock(block *types.Block) error {
-	if block == nil || block.Header == nil || block.Body == nil {
+	if block == nil {
```

Comment:
Before these types were pointers so there was a nil check, but now they aren't pointers. I can still do this check on block.Body, but since the header contains a VDT slice it isn't comparable so I cannot do this nil check. I alluded to this in an above comment, but if there was more to this check than just making sure there isn't a nil pointer reference I would need to discuss how to handle that for the header. Could I get your thoughts on this @noot?

Reply:
Now that you've implemented the `Header.Empty()` receiver func you can use it here and implement a `Block.Empty()` method which checks zero case and/or 0 length `Block.Body` and `Header.Empty()`.